ngx_http_sxg_filter_module.c

```diff
@@ -53,6 +54,8 @@ typedef struct {
   bool cert_chain_mode;
   ngx_str_t link_header;
   ngx_array_t subresource_list;
+
+  pthread_mutex_t lock;
```

Comment:
I don't understand nginx; could you document whether there is one lock per process, per worker, per request, etc.?

Reply:
Per request. Appended comment.